src/clock.js

```diff
@@ -0,0 +1,21 @@
+class Clock {
+  // Return the current time in milliseconds
+  msec () {
+    return new Date().getTime();
+  }
+
+  // Run fn periodically, whether async or not, surviving errors
+  periodically (interval, name, fn) {
+    let iterator = () => {
```

Comment:
You would be better off with:
```js
(async () => {
  while (true) {
    try {
      await fn();
    } catch (err) {
      console.log(err);
    }
    await new Promise(resolve => setTimeout(resolve, interval));
  }
})();
```

it's a lot cleaner and the promise stuff you're doing now won't survive synchronous exceptions from `fn()`